pkg/storages/sh/folder.go

```diff
@@ -268,6 +270,11 @@ func (folder *Folder) PutObject(name string, content io.Reader) error {
 	return nil
 }
 
+func (folder *Folder) PutObjectWithContext(ctx context.Context, name string, content io.Reader) error {
+	ctxReader := contextio.NewReader(ctx, content)
+	return folder.PutObject(name, ctxReader)
```

Comment:
Seems like contextio could be skipped in favor of adding a context parameter to folder.PutObject